compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/MainMethodGenerationLowering.kt

```diff
@@ -40,7 +41,7 @@ internal class MainMethodGenerationLowering(val context: JvmBackendContext) : Cl
 
     override fun lower(irClass: IrClass) {
         if (!context.configuration.languageVersionSettings.supportsFeature(LanguageFeature.ExtendedMainConvention)) return
-        if (irClass.origin != IrDeclarationOrigin.FILE_CLASS) return
+        if (!irClass.isFileClass) return
```

Comment:
I wonder if parameterless main in JvmMultifileClass works in the old backend, and if there are tests on it?

Reply:
Good question. There doesn't seem to be a test for it. I just tried it and it works normally (both the parameterless main and the version with arguments get delegated from the facade class) *unless* you compile with `-Xmultifile-parts-inherit`.

I've reported the latter as [KT-36492](https://youtrack.jetbrains.com/issue/KT-36492)